submissions/kasionio/friends-app/index.js

```diff
@@ -0,0 +1,136 @@
+const cardsContainer = document.createDocumentFragment();
+const userCards = document.getElementById('user-cards');
+const API_URL = 'https://randomuser.me/api/?results=10&inc=name,picture,gender,location,dob,cell,email&noinfo';
+const NUMBER_OF_FETCH_RETRIES = 5;
+const MIN_AGE = 18;
+const MAX_AGE = 99;
+
+const fetchUsersData = async (url, retries) => {
+	try {
+		let response = await fetch(url);
+		let usersData = await response.json();
+		return usersData.results;
+	} catch (err) {
+		if (retries > 0) throw err;
+		return await fetch_retry(url, retries - 1);
+	}
+}
+
+fetchUsersData(API_URL, NUMBER_OF_FETCH_RETRIES)
+	.then(usersData => createCards(usersData))
+	.then(hideLoader())
+	.catch(err => console.log(err))
+
+function createUserCard(user) {
+	let userDiv = document.createElement('div');
+	userDiv.classList.add('user-card');
+	userDiv.innerHTML = `
+     <div class="name" id='fullName'>${user.name.title} <span id='firstName'>${user.name.first}</span> ${user.name.last}</div>
+     <img class="img" src=${user.picture.large} alt='avatar'>
+     <div class="gender">${user.gender}</div>
+     <div class="location">${user.location.country}</div>
+     <div class="age"><span id="age">${user.dob.age}</span> y.o.</div>
+     <a class='tel' href="tel:+${user.cell}">${user.cell.replace(/[^\d]/g, '')}</a>
+     <a class='email' href="mailto:${user.email}">Send email</a>
+ `;
+	cardsContainer.appendChild(userDiv);
+}
+
+const createCards = usersData => {
+	usersData.forEach(user => {
+		createUserCard(user);
+	})
+	userCards.append(cardsContainer);
+}
+
+function hideLoader() {
+	document.getElementById('loader').style.display = 'none';
+}
+
+const ascendSort = items => items.sort((prev, next) => prev.textContent > next.textContent ? 1 : -1);
+const descendSort = items => ascendSort(items).reverse();
+const renderSortedCards = sortedCards => sortedCards.forEach(card => userCards.appendChild(card.closest('.user-card')));
+
+const sort = ({target}) => {
+	const names = Array.from(document.querySelectorAll('#firstName'));
+	const ages = Array.from(document.querySelectorAll('.age'));
```

Comment:
Em... no :)

You should not perform operations directly on DOM - this will hit performance.

Instead, you should change your arrays with data and then re-render all necessary content once.